### PR TITLE
Improving Chat

### PR DESCRIPTION
**Pull request by SirLez**
Improved Chatting:
- Fixed chat bubble glitched
- Fixed WebSocket Failures
- Added a WebSocket Reconnector
- Added Chat Tools
- Added Message Position Implements

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -36,7 +36,8 @@
         </activity>
         <activity
             android:name=".activities.ChatActivity"
-            android:exported="false">
+            android:exported="false"
+            android:screenOrientation="portrait">
```
Comment (4med) [inline app/src/main/AndroidManifest.xml:40]: No idea why this is here

File: app/src/main/java/com/better/amino/activities/ChatActivity.java [hunk 2]
```diff
@@ -55,8 +110,51 @@ private void instantiateWebSocket() {
         OkHttpClient client = new OkHttpClient();
         Request request = new Request.Builder().url("wss://ws1.narvii.com/?signbody=" + data.replace("|", "%7C")).headers(headerBuilder.build()).build();
 
-        MessagesAdapter adapter = new MessagesAdapter(this, new ArrayList<>());
         WebSocket webSocket = client.newWebSocket(request, new MessageListener(this, adapter));
+    }
+
+    private void showMessageTypesPopup() {
```
Comment (4med) [inline app/src/main/java/com/better/amino/activities/ChatActivity.java:116]: Good idea, will make the code split to functions in the future

File: app/src/main/res/drawable/coin.xml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
+    android:viewportWidth="48"
+    android:viewportHeight="48">
+    <path
+        android:fillColor="#FF000000"
```
Comment (4med) [inline app/src/main/res/drawable/coin.xml:7]: Should be surface

**Review by 4med — COMMENTED** [review]
Should edit some things

**Review by SirLez — COMMENTED** [review]
Reviewed

**Review by 4med — APPROVED** [review]

**Pull request merged**